h5netcdf/core.py

```diff
@@ -552,9 +561,20 @@ def resize_dimension(self, dimension, size):
 class File(Group):
 
     def __init__(self, path, mode='a', invalid_netcdf=None, **kwargs):
-        self._preexisting_file = os.path.exists(path)
         try:
-            self._h5file = h5py.File(path, mode, **kwargs)
+            if path.startswith(('http', 'hdf5:')):
+                if no_h5pyd:
+                    raise RuntimeError('h5pyd package required')
```

Comment:
I think the cleanest exception type here would be `IOError`.

Also, let's make the error message more informative:
```
raise ImportError("No module named 'h5pyd'. h5pyd is required for "
                  "opening remote paths with h5netcdf: {}".format(path))
```